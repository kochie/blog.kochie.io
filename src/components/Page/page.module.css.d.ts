declare const styles: {
  readonly page: string
  readonly container: string
}
export = styles
