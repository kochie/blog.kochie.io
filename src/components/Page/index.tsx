import React, { PropsWithChildren, ReactElement } from 'react'

import Footer from '@/components/Footer'
import Topbar from '@/components/Topbar'

import style from './Page.module.css'

const Page = ({ children }: PropsWithChildren<null>): ReactElement => {
  return (
    <div className={style.page}>
      <Topbar />
      <div className={style.container}>{children}</div>
      <Footer
        title={'Kochie Engineering'}
        links={[
          { name: 'me', src: 'https://me.kochie.io', goal: 'SEQGQC1X' },
          {
            name: 'linkedin',
            src: 'https://linkedin.com/in/rkkochie',
            goal: 'RMXXVNIC',
          },
          {
            name: 'rss',
            src: `https://${
              process.env.NEXT_PUBLIC_PROD_URL ||
              process.env.NEXT_PUBLIC_VERCEL_URL
            }/feed/rss.xml`,
            goal: 'PZQY507K',
          },
        ]}
      />
    </div>
  )
}

export default Page
