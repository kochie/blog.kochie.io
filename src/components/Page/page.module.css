.page {
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  overflow: hidden;
}

.container {
  flex-grow: 1;
}