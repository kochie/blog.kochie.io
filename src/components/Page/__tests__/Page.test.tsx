import React from 'react'
import { ReactTestRenderer, act, create } from 'react-test-renderer'
import { library } from '@fortawesome/fontawesome-svg-core'
import { faCopyright } from '@fortawesome/pro-duotone-svg-icons'

import Page from '..'

beforeAll(() => {
  library.add(faCopyright)
})

describe('PAGE COMPONENT', () => {
  test('renders correctly', () => {
    let root: ReactTestRenderer

    act(() => {
      root = create(
        <Page>
          <div>
            <p>This is a test page</p>
          </div>
        </Page>
      )
    })

    // @ts-expect-error tree will be assigned
    expect(root.toJSON()).toMatchSnapshot()
  })
})
