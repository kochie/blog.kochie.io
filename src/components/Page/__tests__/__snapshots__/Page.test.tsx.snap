// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PAGE COMPONENT > renders correctly 1`] = `
<DocumentFragment>
  <div
    class="_page_508746"
  >
    <div
      class="flex flex-row justify-center items-center list-none bg-gray-500 m-0 p-0 fixed w-screen z-40"
    >
      <a
        href="/authors"
      >
        <div
          class="_child_1b1f86 _heading_1b1f86"
        >
          Authors
        </div>
      </a>
      <a
        href="/"
      >
        <div
          class="_child_1b1f86 _heading_1b1f86"
        >
          Articles
        </div>
      </a>
      <a
        href="/tags"
      >
        <div
          class="_child_1b1f86 _heading_1b1f86"
        >
          Tags
        </div>
      </a>
    </div>
    <div
      class="_container_508746"
    >
      <div>
        <p>
          This is a test page
        </p>
      </div>
    </div>
    <div
      class="w-full bg-gray-500"
    >
      <div
        class="flex flex-col justify-evenly md:content-evenly items-center px-4 text-white h-20 md:m-auto md:flex-row md:justify-between md:align-baseline lg:max-w-5xl"
      >
        <p>
          Kochie Engineering 
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-copyright _copyright_acfd2c"
            data-icon="copyright"
            data-prefix="fad"
            role="img"
            viewBox="0 0 512 512"
          >
            <g
              class="fa-duotone-group"
            >
              <path
                class="fa-secondary"
                d="M0 256a256 256 0 1 0 512 0 256 256 0 1 0 -512 0zm171.1-84.9c46.9-46.9 122.8-46.9 169.7 0 9.4 9.4 9.4 24.6 0 33.9s-24.6 9.4-33.9 0c-28.1-28.1-73.7-28.1-101.8 0s-28.1 73.7 0 101.8 73.7 28.1 101.8 0c9.4-9.4 24.6-9.4 33.9 0s9.4 24.6 0 33.9c-46.9 46.9-122.8 46.9-169.7 0s-46.9-122.8 0-169.7z"
                fill="currentColor"
              />
              <path
                class="fa-primary"
                d="M306.9 205.1c-28.1-28.1-73.7-28.1-101.8 0s-28.1 73.7 0 101.8 73.7 28.1 101.8 0c9.4-9.4 24.6-9.4 33.9 0s9.4 24.6 0 33.9c-46.9 46.9-122.8 46.9-169.7 0s-46.9-122.8 0-169.7 122.8-46.9 169.7 0c9.4 9.4 9.4 24.6 0 33.9s-24.6 9.4-33.9 0z"
                fill="currentColor"
              />
            </g>
          </svg>
           2020
        </p>
        <div
          class="flex relative"
        >
          <div
            class="relative ml-4"
          >
            <a
              class="_heading_acfd2c"
              href="https://me.kochie.io"
            >
              me
            </a>
          </div>
          <div
            class="relative ml-4 _divider_acfd2c"
          >
            <a
              class="_heading_acfd2c"
              href="https://linkedin.com/in/rkkochie"
            >
              linkedin
            </a>
          </div>
          <div
            class="relative ml-4 _divider_acfd2c"
          >
            <a
              class="_heading_acfd2c"
              href="https://undefined/feed/rss.xml"
            >
              rss
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
